sensu/plugins/check-keystone-expired-tokens.rb

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env /opt/sensu/embedded/bin/ruby
+#
+# Openstack Keystone expired token Plugin
+# ===
+#
+# This plugin checks the number of expired token in Keystone and warns you according to specified limits
+#
+# Released under the same terms as Sensu (the MIT license); see LICENSE
+# for details.
+
+
+require 'rubygems' if RUBY_VERSION < '1.9.0'
+require 'sensu-plugin/check/cli'
+
+class CheckKeystoneExpiredToken < Sensu::Plugin::Check::CLI
+
+  option  :warning,
+          :description => "warning count of expired token",
+          :short => '-w NUMBER',
+          :long => '--warning NUMBER',
+          :proc => proc {|a| a.to_i },
+          :default => 1000
+		  
+  option  :critical,
+          :description => "critical count of expired token",
+          :short => '-c NUMBER',
+          :long => '--critical NUMBER',
+          :proc => proc {|a| a.to_i },
+          :default => 10000
+
+  option  :criticality,
+          :description => "Set sensu alert level, default is critical",
+          :short => '-z CRITICALITY',
+          :long => '--criticality CRITICALITY',
+          :default => 'critical'
+
+  def run
+    expired_token_count = `mysql -u root -N -s -e "select count(*) from keystone.token where expires<now()"`
```

Comment:
we shouldn't assume root access ... we should ask the user to pass username and password in.  That way if we wanted to we could create a sensu specific user in mysql with very limited permissions.